DiagnosticsExtension/Controllers/ConnectionStringValidationController.cs

```diff
@@ -0,0 +1,86 @@
+//-----------------------------------------------------------------------
+// <copyright file="UdpEchoTestController.cs" company="Microsoft Corporation">
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License. See LICENSE in the project root for license information.
+// </copyright>
+//-----------------------------------------------------------------------
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Net.Sockets;
+using System.Text;
+using System.Threading.Tasks;
+using System.Web.Http;
+using DiagnosticsExtension.Models.ConnectionStringValidator;
+
+namespace DiagnosticsExtension.Controllers
+{
+    /// <summary>
+    /// Worker instances are running an udp echo server on port 30000. This controller is for checking the connection between target 
+    /// worker instance by pinging and checking the echoed result.
+    /// </summary>
+    [RoutePrefix("api/connectionstringvalidation")]
+    public class ConnectionStringValidationController : ApiController
+    {
+        [HttpGet]
+        [Route("test")]
+        public async Task<HttpResponseMessage> Test(string connStr, int? typeId = null)
```

Comment:
nit: Recommending a more specific function and route name:
Route("test-connection-string")
TestConnectionString()

Reply:
The controller has a route prefix `api/connectionstringvalidation`, so for this route it's `api/connectionstringvalidation/test`

Reply:
changed to `api/connectionstringvalidation/validate`